MatrixKit/Controllers/MXKAttachmentsViewController.m

```diff
@@ -1113,6 +1107,14 @@ - (void)scrollViewDidEndDecelerating:(UIScrollView *)scrollView
             [self refreshCurrentVisibleCell];
         }
     }
+    
+    //tell the delegate that a new Attachment has been shown and pass eventId
+    MXKMediaCollectionViewCell* cell = [[self.attachmentsCollection visibleCells] firstObject];
+    NSString *attachmentEventId = ((MXKAttachment *)attachments[cell.tag]).eventId;
```

Comment:
Warning: cell.tag may be -1 here, if the current cell is the one with the spinner displayed during the back pagination (isBackPaginationInProgress == YES).

1- I suggest moving the "new attachment notification" inside the `refreshCurrentVisibleCell` method. Then you will be able to use the actual visible attachment.
2- You should call `displayedNewAttachmentWithEventId` with a nil event id, when the current cell is the pagination spinner ((isBackPaginationInProgress == YES) && (currentVisibleItemIndex == 0))